server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1592,8 +1592,11 @@ private void executeOneStaleIndexDelete(
                                 Map<String, BlobMetadata> shardLevelBlobs = shardBlob.getValue().listBlobs();
                                 for (Map.Entry<String, BlobMetadata> shardLevelBlob : shardLevelBlobs.entrySet()) {
                                     String blob = shardLevelBlob.getKey();
-                                    String snapshotUUID = blob.substring(SHALLOW_SNAPSHOT_PREFIX.length(), blob.length() - ".dat".length());
                                     if (blob.startsWith(SHALLOW_SNAPSHOT_PREFIX) && blob.endsWith(".dat")) {
+                                        String snapshotUUID = blob.substring(
```

Comment:
Looks like this code is duplicated on L#3365. Would it make sense to pull it out into a common function?
I know the logic will not change frequently enough, but a common location will ensure consistency.

Reply:
Updated. Made a few other minor/cosmetic changes as well. Note the diff is much easier to follow if you hide whitespace changes as I removed an outer condition that changed the indent.